### PR TITLE
GitHub sync

### PR DESCRIPTION
**Pull request by nshahan**
Includes fixes for compatibility with Dart SDK v2.0.0-dev.62.0.

Update example branch:
https://github.com/dart-lang/angular_components/tree/github-sync

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,40 @@
...
+   * Add `role` attribute.
+   * Add `material-radio-color` mixin and deprecate `material-radio-theme`.
+ * Add DI modules to Laminate Overlay & Popup components.
+ * Update to `SelectionModel.single()`, `SelectionModel.multi()` etc. instead of
+   deprecated `SelectionModel.withList()`.
+ * Remove explicit `Visibility.local` from Angular components and directives, as
+   this is the default now.
+ * Remove legacy CSS class names used by `ShowHideDirective` that conflict with
+   Angular JS apps.
+ * Style updates for new additions to Effective Dart.
```
Comment (chalin) [inline CHANGELOG.md]: > Style updates for new additions to Effective Dart.

-> (maybe)

> Updates to conform to new Effective Dart guidelines.

File: lib/focus/focus_interface.dart [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// A component or directive that can be programmatically focused.
+///
+/// Directive can manage if it means to put focus on root of itself
```
Comment (chalin) [inline lib/focus/focus_interface.dart]: "Directive can manage if it means to ..." I'm not quite sure what that means. Maybe just suggest that "focus is put on this component or directive, or meaningful component inside."

File: lib/laminate/overlay/module.dart [hunk 3]
```diff
@@ -77,27 +77,52 @@ HtmlElement getOverlayContainerParent(Document document,
   return containerParent ?? document.querySelector('body');
 }
 
-/// DI bindings for Overlay and its dependencies.
-const overlayBindings = const [
-  AcxImperativeViewUtils,
-  DomRuler,
+/// DI module for Overlay and its dependencies.
+const overlayModule = const Module(
```
Comment (chalin) [inline lib/laminate/overlay/module.dart]: Oh, Modules! Nice :)

File: lib/material_menu/menu_popup_wrapper.dart [hunk 4]
```diff
@@ -28,15 +28,10 @@ class MenuPopupWrapper implements AcceptsWidth {
   Stream<bool> get isExpandedChange => _expanded.stream;
   final _expanded = new ObservableReference<bool>(false);
 
-  /// Width of the menu.
+  /// Selects 1 of 5 predefined width values for the menu.
```
Comment (chalin) [inline lib/material_menu/menu_popup_wrapper.dart]: 👍

File: lib/material_menu/menu_popup_wrapper.dart [hunk 5]
```diff
@@ -28,15 +28,10 @@ class MenuPopupWrapper implements AcceptsWidth {
...
   ///
-  /// Ignored if [menu] instance has a width set.
-  /// Preset width, 1 through 5. By default, the material list will expand to
-  /// its content. Note: The spec clearly lays out predefined list sizes so use
-  /// the default, expanding size, sparingly.
-  /// Each width multiplies the base block width (64px on desktop and tablet) by
-  /// [1.5, 3, 5, 6, 7], respectively to obtain a predictable width.
-  /// Set to 0 to have the list expand to fit its content.
+  /// Ignored if [menu] instance has a width set. See [AcceptsWidth.width] for
+  /// more details.
```
Comment (chalin) [inline lib/material_menu/menu_popup_wrapper.dart]: 👍

File: lib/material_progress/_mixins.scss [hunk 6]
```diff
@@ -29,3 +29,22 @@
...
   }
 }
+
+// Changes the height and roundedness of the ends of <material-progress>.
+//
+// Example use:
+// :host material-progress {
+//   wide-rounded-progress-bar($mat-grid * 2);
+// }
+//
```
Comment (chalin) [inline lib/material_progress/_mixins.scss]: Not sure we actually process "doc comments", but you're using `///` elsewhere, maybe you want that here too?

File: lib/material_slider/material_slider.dart [hunk 7]
```diff
@@ -24,22 +25,36 @@ import 'package:angular_components/utils/browser/dom_service/dom_service.dart';
 ///     <material-slider [min]="0"
 ///                      [max]="100"
 ///                      [step]="1"
-///                      [(value)]="30">
+///                      [(value)]="30"
+///                      [disabled]="disabled">
 ///     </material-slider>
```
Comment (chalin) [inline lib/material_slider/material_slider.dart]: nit: use fenced code formatting (and style it as `html`).

File: lib/src/laminate/popup/popup_source_directive.dart [hunk 8]
```diff
@@ -105,4 +108,13 @@ class PopupSourceDirective
...
   }
+
+  @override
+  void focus() {
+    if (_focusable != null) {
+      _focusable.focus();
+    } else {
+      _element.focus();
+    }
+  }
```
Comment (chalin) [inline lib/src/laminate/popup/popup_source_directive.dart]: Maybe:

```dart
void focus() => (_focusable ?? _element).focus();
```

File: lib/material_progress/_mixins.scss (hunk 6 above)
Comment (nshahan) [reply lib/material_progress/_mixins.scss]: We are planning on adding some doc comment like generation tool. There will need to be a cleanup fixit when we have something to build and look at.

**Pull request merged**